server/utils/formUtils.ts

```diff
@@ -175,11 +175,17 @@ export function convertKeyValuePairsToSummaryListItems<T extends object>(
 export const summaryListItem = (
   label: string,
   value: string,
-  renderAs: keyof TextItem | keyof HtmlItem = 'text',
-): SummaryListItem => ({
-  key: { text: label },
-  value: renderAs === 'text' ? { text: value } : { html: value },
-})
+  renderAs: keyof TextItem | keyof HtmlItem | 'textBlock' = 'text',
+  supressBlank = false,
+): SummaryListItem => {
+  const htmlValue = renderAs === 'textBlock' ? `<span class="govuk-summary-list__textblock">${value}</span>` : value
+  return !supressBlank || value
+    ? {
+        key: { text: label },
+        value: renderAs === 'text' ? { text: value } : { html: htmlValue },
+      }
+    : undefined
+}
```

Comment:
I'm thinking that this routine could be expanded to include common formats such as date and times too. Also, not sure what the default behaviour should be wrt including rows with falsey values. I think they are mostly suppressed but the default is to include them.

Reply:
Yes, that's a very good call -- I like the idea of value types other than `text` or `html`. Maybe a wrapper function setup to handle the hiding of falsey values would work? We would probably need to ensure any value renderer would definitely return nothing when no data -- I seem to remember some of the lists are return as empty `<ul/>` -- this wouldn't work!